### ISSUE TITLE
Refactor the Navigation block > Settings > DIsplay UI for better usability and accessibility

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59793]
### Description

The user interface for the Navigation block 'Display' settings has room for improvements for better usability and accessibility.

Screenshot of the current UI in 3 different states:

![Screenshot 2024-03-12 at 15 36 18](https://github.com/WordPress/gutenberg/assets/1682452/541f84a9-cbbd-4f1b-a2ba-3644be8c20c1)

- By default, the first setting in the 'DIsplay' section is the Overlai Menu setting, which is set to Off by default.
- When switching this setting to Mobile or Always, new settings appear _before_ it.
- This is less than ideal of all users as the normak expectation is that something appears _after_.
- It is even more important for keyboard users and screen reader users as navigating backward with Shift+Tab isn't intuitive to discover new things. Screen reader users may totally miss that something appeared _before_ the controls they just activated.
- The new section that appears has no title. It's only visuals and identified by two icons inside a grey box:

![Screenshot 2024-03-12 at 15 56 01](https://github.com/WordPress/gutenberg/assets/1682452/77084fef-de53-487b-937b-e3b579f145f3)

- This grey box is actually a button labeled 'Overlay menu controls'. There is no text of tooltip to visually expose the button name. It is worth reminding, once again, that _all icon buttons must at least expoe their accessible name by the means of a tooltip_.
- Clicking this byutton expands a new section that contains:
  - A 'Show icon button' toggle.
  - A radio buttons group to choose the menu toggle icon. This radio group looks like two buttons but, under the hood, it's semantically two radio buttons.
- Radio buttons should always be grouped within a `fieldset` element to clarify what they are meant to.
- The two radio buttons only use icons: there is no text or tooltip to visually expose their accessible name. Again, all icon buttons should at least use a tooltip.
- The two radio buttons are labelled, lowercase: 'handle' and 'menu'. There is no other textual information to explain these two options actually set the icon type used for the menu toggle.
  - Screen readers will announce these radio buttons iierally as 'handle' and 'menu', which isn't any helpful.
  - Speech recognition software users will have no clue how to activate these radio buttons because their accessible name isn't visually exposed.
  - Users with cognitive load or non-tech-savvy users will have hard times figuring out what these radio buttons do.

Additionally, I'm not sure implementing custom UIs and 'ad hoc' designs is the best way to build a consistent, predictable, and usable interface Gutenberg is based on a library of reusable components. Ad hoc implementations shoul dbe avoided in favor of the reusable base components. If, for any reason, a variant of a base component is desired and does have a good use case that isn't a 'unique' case, then a new variant of the base compoennt should be proposed, discussed, implemented, and used to provide the most usable and accessible experience for everyone.

### Step-by-step reproduction instructions

- Go to the Site editor and select a navigation block that does have a menu.
- In the block inspector panel, go to the Settings tab > Display.
- Click Overlay Menu > Mobile or Always.
- Observe a new grey box appears _before_ the Overlay Menu section. Expected: to appear after it.
- Observe the grey box that appeared doesn't really look like an expandable panel.
- Observe the grey box doesn't show any text or title to explain what it is. It only contains two icons.
- Observe one of the icons is an X close icon, which is confusing because users may think it's a control to 'close' the grety box.
- Click the gray box.
- Observe new controls appear.
- Hover or focus the two buttons after the title 'Icon'.
- Observe these two buttons don't have any text or tooltip.
- Inspect the source or use a screen reader and observe these two buttons are labeled 'handle' and 'menu', which really doesn't explain what they are meant to do.

Additionally:
Observe the related code at https://github.com/WordPress/gutenberg/blob/f6531c3a708f7aade1f9b6c6711cfa8c1da3e89d/packages/block-library/src/navigation/edit/overlay-menu-preview.js#L36-L45

Observe the two translatable strings: `__( 'handle' )` and `__( 'menu' )`. They don't have any translators comment or context provided by the means of the function `_x()` to explain translators how to better translate them.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by getdave** [comment]
Thank you for the Issue. To make it more actionable, I think it would benefit from being broken down into discreet sub Issues that can be addressed incrementally.

I appreciate the original reporter may not have time/inclination to do that. I will try and circle back to help out.
